File: member-service/src/main/java/com/hedvig/memberservice/commands/StartOnboardingWithSSNCommand.java
package com.hedvig.memberservice.commands;

import com.hedvig.memberservice.web.dto.StartOnboardingWithSSNRequest;
import lombok.Value;
import org.axonframework.commandhandling.TargetAggregateIdentifier;

//This is deprecated and will only work for sweden should removet once bot-service is not used for on boarding
@Value
@Deprecated
public class StartOnboardingWithSSNCommand {
  @TargetAggregateIdentifier long memberId;
  String ssn;

  public StartOnboardingWithSSNCommand(long memberId, StartOnboardingWithSSNRequest requestData) {
    this.memberId = memberId;
    this.ssn = requestData.getSsn();
  }
}

File: member-service/src/main/java/com/hedvig/memberservice/commands/StartOnboardingWithSSNCommand.kt
package com.hedvig.memberservice.commands

import com.hedvig.memberservice.web.dto.StartOnboardingWithSSNRequest
import org.axonframework.commandhandling.TargetAggregateIdentifier

@Deprecated("This is deprecated and will only work for sweden should remove once bot-service is not used for on boarding")
class StartOnboardingWithSSNCommand(
    @TargetAggregateIdentifier val memberId: Long,
    requestData: StartOnboardingWithSSNRequest
) {
    var ssn: String

    init {
        ssn = requestData.ssn
    }
}
